sql/core/src/main/scala/org/apache/spark/sql/json/JSONRelation.scala

```diff
@@ -105,6 +107,15 @@ private[sql] class JSONRelation(
     jsonSchema
   }
 
+  override private[sql] def buildScan(
+      requiredColumns: Array[String],
+      filters: Array[Filter],
+      inputPaths: Array[String],
+      broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
I don't mean we need to refresh the schema / metadata, but fetching the latest file under the table path.